### PR TITLE
6392 batch action bug

### PR DESCRIPTION
**Pull request by arisah-ijeoma**

### CODE REVIEW

File: app/assets/javascripts/backbone/views/timeline_select_steps_view.coffee [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class MS.Views.TimelineSelectStepsView extends Backbone.View
 
+  initialize: ->
+    # on page load, disable the batch actions since no step would have been checked
+    $('.dropdown-menu.batch-actions > li').addClass("disabled");
```
Comment (emitche) [inline app/assets/javascripts/backbone/views/timeline_select_steps_view.coffee]: Smart.

File: app/assets/javascripts/backbone/views/timeline_select_steps_view.coffee [hunk 2]
```diff
@@ -57,3 +61,9 @@ class MS.Views.TimelineSelectStepsView extends Backbone.View
         ids.push($(@).data('id'))
 
     @$('.step-ids').attr('value', ids)
+
+    # disable/enable the batch actions depending on if any step is checked
+    if ids.length == 0
```
Comment (emitche) [inline app/assets/javascripts/backbone/views/timeline_select_steps_view.coffee]: :) I like this approach.

File: app/controllers/admin/project_steps_controller.rb [hunk 3]
```diff
@@ -123,6 +123,9 @@ def duplicate
   end
 
   def batch_destroy
+    # don't initialize batch destroy if there are no step ids
```
Comment (emitche) [inline app/controllers/admin/project_steps_controller.rb]: The UI should catch any cases when there are no steps selected. Are the changes in this file just a safeguard?

**Comment by emitche** [comment]
Looks good. I just have one question.

File: app/controllers/admin/project_steps_controller.rb (hunk 3 above)
Comment (arisah-ijeoma) [reply app/controllers/admin/project_steps_controller.rb]: Yep
The UI would catch those due to the frontend manipulation but this safeguards our backend and makes sure no dirty data gets in, especially the wrong way

**Pull request merged**